src/test/java/net/snowflake/client/jdbc/RestRequestWiremockTest.java

```diff
@@ -0,0 +1,250 @@
+    }
+
+    private static String getWiremockStandAlonePath() {
+        return System.getProperty("user.home") + WIREMOCK_M2_PATH;
+    }
+
+    private static String getResourceURL(String relativePath) {
+        return Paths.get(systemGetProperty("user.dir"), "src", "test", "resources", relativePath)
+                .toAbsolutePath()
+                .toString();
+    }
+
+    private static void waitForWiremock() {
+        await()
+                .pollDelay(Duration.ofSeconds(1))
+                .atMost(Duration.ofSeconds(3))
+                .until(RestRequestWiremockTest::isWiremockResponding);
+    }
+
+    private static boolean isWiremockResponding() {
+        try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+            HttpGet request =
+                    new HttpGet(String.format("http://%s:%d/__admin/mappings", WIREMOCK_HOST, httpProxyPort));
+            CloseableHttpResponse response = httpClient.execute(request);
+            return response.getStatusLine().getStatusCode() == 200;
+        } catch (Exception e) {
+            logger.warn("Waiting for wiremock to respond: ", e);
+        }
+        return false;
+    }
+
+    private static void stopWiremockStandAlone() {
+        if (wiremockStandalone != null) {
+            wiremockStandalone.destroyForcibly();
+            await()
+                    .alias("stop wiremock")
+                    .atMost(Duration.ofSeconds(10))
+                    .until(() -> !wiremockStandalone.isAlive());
+        }
+    }
+
+    @BeforeClass
+    public static void setUpClass() {
+        assumeFalse(RunningNotOnJava8.isRunningOnJava8());
+        assumeFalse(RunningNotOnJava21.isRunningOnJava21());
+        assumeFalse(
+                RunningNotOnGithubActionsMac
+                        .isRunningOnGithubActionsMac()); // disabled until issue with access to localhost
+        // (https://github.com/snowflakedb/snowflake-jdbc/pull/1807#discussion_r1686229430) is fixed on
+        // github actions mac image. Ticket to enable when fixed: SNOW-1555950
+        originalTrustStorePath = systemGetProperty(TRUST_STORE_PROPERTY);
+        startWiremockStandAlone();
+    }
+
+    private static void addMapping(String mapping) {
+        try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+            HttpPost request =
+                new HttpPost(String.format("http://%s:%d/__admin/mappings/import", WIREMOCK_HOST, httpProxyPort));
+
+            request.setEntity(new StringEntity(mapping));
+            CloseableHttpResponse response = httpClient.execute(request);
+            if (response.getStatusLine().getStatusCode() != 200) {
+                System.out.println(response.getStatusLine().getStatusCode());
+                System.out.println(EntityUtils.toString(response.getEntity()));
+            }
+            assumeTrue( response.getStatusLine().getStatusCode() == 200);
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
and here also I would use logger and said maybe that setting wiremock mapping failed so it is more self explanatory in comparison to plain stack trace